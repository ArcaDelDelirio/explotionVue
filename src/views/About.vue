<template>
  <div class="about">
    <NavData />
    <h1>List of Movies</h1>
    <button @click="getAllMovies">All movies</button>
    <div v-if="dataApi.hasOwnProperty('0')">
      {{dataApi}}
    </div>
  </div>
</template>
<script>
// @ is an alias to /src
import NavData from '@/components/NavData.vue'

export default {
  name: 'about',
  components: {
    NavData,
  },
  data(){
    return {
      dataApi:{

      }
    }
  },
  methods: {
    getAllMovies(){
      return fetch(' https://damp-falls-68291.herokuapp.com/movies/get-movies')
              .then(res => res.json())
              .then(response => this.dataApi = response)
              .catch(error => this.dataApi = error)
    }
  }
}
</script>