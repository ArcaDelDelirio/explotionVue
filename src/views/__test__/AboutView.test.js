import { mount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import About from '../About';

describe('About View', () => {
  it('Render the component', () => {
    const localVue = createLocalVue();
    localVue.use(VueRouter);
    const router = new VueRouter();
    const wrapper = mount(About, {
      localVue,
      router,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
