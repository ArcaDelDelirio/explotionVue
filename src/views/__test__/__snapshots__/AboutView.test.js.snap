// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About View Render the component 1`] = `
<div class="about">
  <header>
    <nav class="navbar">
      <div class="logo-container"><span class="logo"><a href="#/" class="router-link-exact-active router-link-active"><img src="@/assets/movie.png" height="24px" width="24px" alt="logo"></a></span> <span>Super Movie DataBase</span></div>
      <div class="login-container">
        <div class="login-state">
          <div class="login-caps"><button class="sign-button">Sign In</button></div>
          <!---->
        </div>
      </div>
    </nav>
  </header>
  <h1>List of Movies</h1> <button>All movies</button>
  <!---->
</div>
`;
